File: 0268-MissingNumbers/MissingNumbers.go
package missingnumbers

func missingNumber(nums []int) int {
	checkList := make([]bool, len(nums)+1)

	for _, val := range nums {
		checkList[val] = true
	}

	for i := range checkList {
		if checkList[i] != true {
			return i
		}
	}

	return -1
}

File: 0268-MissingNumbers/MissingNumbers.java

class MissingNumbers {
    class Solution {
        public int missingNumber(int[] nums) {
            boolean[] checkList = new boolean[nums.length + 1];
            for(int i = 0 ; i < nums.length; i++){
                checkList[nums[i]] = true;
            }
            
            for(int i = 0; i < checkList.length; i++){
                if(!checkList[i]) return i;
            }
            
            return -1;
        }
    }
}
